lib/system-pkgs.nix

```diff
@@ -12,10 +12,10 @@ pkgs:
   # Base packages.
   pkgs
 
-  # fetchgit should always come from the evalPackages
+  # fetchgit should always come from the buildPackages
   # if it comes from the targetPackages we won't even
   # be able to execute it.
-  // { fetchgit = pkgs.evalPackages.fetchgit; }
+  // { fetchgit = pkgs.buildPackages.fetchgit; }
```

Comment:
I checked and it does look like the problem with `gitMinimal` has been fixed in nixpkgs 22.05.  I think we should keep this fix for another year or so for people using older `nixpkgs`.  I've moved this to `overlays/nix-prefetch-git-minimal.nix`.  I also included a better comment and the `nix-diff` command I used for checking `nixpkgs`.

Reply:
This turned out to be a small [bug in the libredirect derivation in nixpkgs](https://github.com/NixOS/nixpkgs/pull/182143).  I've put an override of libredirect in the overlay instead of messing with `fetchgit`.